Feature/7720 my store summary screen

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7720
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
Adds My store summary screen UI. This screen is presented right after domain picker step, whether by clicking "Continue" or by clicking "Skip" on domain picker step. 

### Testing instructions
Check that the design matches Figma designs HyVloP5FipZzyPVenH2euI-fi-1980%3A94456

### Images/gif

| Light  | Dark |
| ------------- | ------------- |
| <img width="300"  src="https://user-images.githubusercontent.com/2663464/201371524-7670bda4-5bb9-443a-b746-878ffe0045c0.png">  | <img width="300"  src="https://user-images.githubusercontent.com/2663464/201373499-48d629c2-7580-4811-9e8b-c9b51da81d4b.png">  |

JorgeMucientes committed:
Add new my store fragment screen after domain picker step

JorgeMucientes committed:
Fix text color from domain picker screen

JorgeMucientes committed:
Add UI for my store summary screen

JorgeMucientes committed:
Add a class data holder for new store input data

JorgeMucientes committed:
Add missing @AndroidEntryPoint

JorgeMucientes committed:
Fix warnings

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt:
```diff
@@ -0,0 +1,26 @@
+import javax.inject.Singleton
+
+@Singleton
+class NewStore @Inject constructor() {
```
@0nko I added this new class as data holder for store creation input data. I've exposed it as a stateflow as it works pretty well with the different viewmodels we are creating for the flow. 

Let me know what you think about it. Furthermore, I'm not super happy with the naming so feel free to suggest and alternative.

0nko replied:
I can't think of a name either.. Let's keep it for now and we can change it later 🤷

0nko reviewed (commented):
The screen looks good, I just noticed the Figma design contains the flag (though I'm not sure if it's been updated - the emoji was supposed to be removed based on the comment below). Also, the button title is slightly different.

![image](https://user-images.githubusercontent.com/1522856/201982066-ca79c0c0-f031-4fc5-bbea-6bb23efa3a2c.png)

JorgeMucientes committed:
Update button with correct copy

JorgeMucientes committed:
Merge branch 'trunk' into feature/7720-my-store-summary-screen

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/domainpicker/DomainPickerFragment.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/domainpicker/DomainPickerViewModel.kt
#	WooCommerce/src/main/res/navigation/nav_graph_store_creation.xml
#	WooCommerce/src/main/res/values/strings.xml

JorgeMucientes committed:
Fix navigation after merging with trunk to include PlansFragment

JorgeMucientes commented:
Thanks for the review and the good catches @0nko 

- I've fixed the copy for the continue button following Figma designs
- Regarding the emoji. Currently, we don't have emojis for countries. Until the API of the profiler step with the list of countries and their emojis is ready, we won't display an emoji. In fact, for M2, the country, and the category fields won't be displayed in this screen as they won't be available. 
- I've added a few fixes in navigation to work well with the recently merged Plans screen

Ready for another round :)

0nko reviewed (approved):
Looks great! 👍 Thanks for the updates.
